src/components/PeopleFilters.tsx

```diff
@@ -1,12 +1,79 @@
+import cn from 'classnames';
+import { useSearchParams, Link } from 'react-router-dom';
+import { getSearchWith, SearchParams } from '../utils/searchHelper';
+
+const SEXES = {
+  m: 'Male',
+  f: 'Female',
+};
+const CENTURIES = ['16', '17', '18', '19', '20'];
+
 export const PeopleFilters = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+  const sex = searchParams.get('sex') || '';
+
+  const setSearchWith = (params: SearchParams) => {
+    const search = getSearchWith(searchParams, params);
+
+    setSearchParams(search);
+  };
+
+  const onSexChange = (newSex: string | null) => {
+    if (sex === null) {
+      return getSearchWith(searchParams, { newSex: null });
+    }
+
+    return getSearchWith(searchParams, { sex: newSex });
+  };
+
+  const onQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchWith({ query: event.target.value || null });
+  };
+
+  const onCenturiesChange = (century: string | null) => {
+    if (century === null) {
+      return getSearchWith(searchParams, {
+        centuries: null,
+      });
+    }
+
+    return getSearchWith(searchParams, {
+      centuries: centuries.includes(century)
+        ? centuries.filter(ch => ch !== century)
+        : [...centuries, century],
+    });
+  };
+
+  const onResetAll = () => {
+    return getSearchWith(new URLSearchParams(), {});
+  };
+
   return (
     <nav className="panel">
       <p className="panel-heading">Filters</p>
 
       <p className="panel-tabs" data-cy="SexFilter">
-        <a className="is-active" href="#/people">All</a>
-        <a className="" href="#/people?sex=m">Male</a>
-        <a className="" href="#/people?sex=f">Female</a>
+        <Link
+          to={{ search: onSexChange(null) }}
+          className={cn({
+            'is-active': !sex,
+          })}
+        >
+          All
+        </Link>
+
+        {Object.entries(SEXES).map(([param, text]) => (
```

Comment:
please don't call a function from the template because it will be executed every time when the state change, so make sense to move it to the variable